src/windows/osd.c

```diff
@@ -371,9 +371,14 @@ ssize_t sendmsg(SOCKET sd, struct msghdr *msg, int flags)
 		offset += msg->msg_iov[i].iov_len;
 	}
 
-	/* send data */
-	sent = sendto(sd, buffer, (int)len, flags,
-		(struct sockaddr *)msg->msg_name, msg->msg_namelen);
+	if (msg->msg_name && msg->msg_namelen) {
+		/* send data over unconnected socket */
+		sent = sendto(sd, buffer, (int)len, flags,
+			(struct sockaddr *)msg->msg_name, msg->msg_namelen);
+	} else {
+		/* send data over connected socket */
+		sent = send(sd, buffer, (int)len, flags);
+	}
```

Comment:
We either need to drop this entire abstraction or rework it.  By having this, we're giving developers the false sense that it's a reasonably usable abstraction on windows.  But it's not.  It allocated a memory buffer and copies the entire message to a new buffer.

If we can't convert the call to iterating over the iovec and calling send/sendto, we need to drop this and push the problem back into the calling location, so that the entire approach can be redone.